delivery/media/dash/lib/media-delivery-dash-parser.js

```diff
@@ -0,0 +1,2006 @@
+            attributePrefix: "@",
+            useDoubleQuotes: !0,
+            arrayAccessFormPaths: DashConstants.ALWAYS_ARRAY_ELEMENTS,
+            nativeTypeAttributes: !0
+        });
+        this.mpd = x2js.xml_str2json(mpdXml);
+    }
+    getJSON() {
+        return this.mpd;
+    }
+    setJSON(mpdJson) {
+        this.mpd = mpdJson;
+    }
+    getMPD() {
+        return new X2JS({
+            attributePrefix: "@",
+            useDoubleQuotes: !0,
+            space: this.indent
+        }).json2xml_str(this.mpd);
+    }
+    setIndent(space) {
+        this.indent = space;
+    }
+    attr(name) {
+        return "@" + name;
+    }
+    filterVideoRenditionByBandwidth(ranges) {
+        function filterFnForVideoAdaptationSet(element) {
+            let result = !1;
+            return ranges.forEach((range => {
+                range[0] <= element[Attr$1(DashConstants.ATTR_BANDWIDTH)] && element[Attr$1(DashConstants.ATTR_BANDWIDTH)] <= range[1] && (result = !0);
+            })), result;
+        }
+        function filterFnForUnknownAdaptationSet(element) {
+            if (!(element[Attr$1(DashConstants.ATTR_MIME_TYPE)] ? element[Attr$1(DashConstants.ATTR_MIME_TYPE)] : "").startsWith("video/")) return !0;
+            let result = !1;
+            return ranges.forEach((range => {
+                range[0] <= element[Attr$1(DashConstants.ATTR_BANDWIDTH)] && element[Attr$1(DashConstants.ATTR_BANDWIDTH)] <= range[1] && (result = !0);
+            })), result;
+        }
+        this.mpd[DashConstants.MPD][DashConstants.PERIOD].forEach((period => {
+            period[DashConstants.ADAPTATION_SET].forEach((adaptationSet => {
+                const _contentType = adaptationSet[this.attr(DashConstants.ATTR_CONTENT_TYPE)] ? adaptationSet[this.attr(DashConstants.ATTR_CONTENT_TYPE)] : "", _mimeType = adaptationSet[this.attr(DashConstants.ATTR_MIME_TYPE)] ? adaptationSet[this.attr(DashConstants.ATTR_MIME_TYPE)] : "";
+                if ("video" == _contentType || _mimeType.startsWith("video/")) {
+                    let filteredRenditions = adaptationSet[DashConstants.REPRESENTATION].filter(filterFnForVideoAdaptationSet);
+                    adaptationSet[DashConstants.REPRESENTATION] = filteredRenditions;
+                } else if ("" === _contentType && "" === _mimeType) {
+                    let filteredRenditions = adaptationSet[DashConstants.REPRESENTATION].filter(filterFnForUnknownAdaptationSet);
+                    adaptationSet[DashConstants.REPRESENTATION] = filteredRenditions;
+                }
+            }));
+        }));
+    }
+}, Attr = function(name) {
+    return "@" + name;
+}, MIME_TYPE_VIDEO_PREFIX = "video/", MIME_TYPE_AUDIO_PREFIX = "audio/", MIME_TYPE_TEXT_PREFIX = "application/";
+
+class DashParser {
+    static parseMPD(mpdXml) {
+        try {
+            dashMpd.parse(mpdXml);
+        } catch (error) {
+            throw new Error(`DashParser: failed to parseMPD due to ${error.message}`);
```

Comment:
@pjaiprakakamai 
> regarding 'Why catch and rethrow the error. Dropping the original exception means the customer can't see the stack trace.'
> I'm not dropping the original exception, just printing the same with a prefix 'DashParser: failed to parseMPD due to' . I'm calling parseMPD from my imported module and would like to catch possible errors that the parser may throw.

Doesn't that drop the stack frames from the original error? AFAIU the message is the string passed into the exception when it was constructed, so creating a new exception hides where it came from. 

The [optional cause parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#rethrowing_an_error_with_a_cause) looks like it would keep the frames.

Reply:
will make this change.